### ISSUE TITLE
FirebirdSampleApplication not working - unsuccessful metadata update

### ISSUE THREAD

**Issue by Timothyoverton** [DbUp/DbUp#243]
Hi Guys,

I'm just trying to get the sample firebird application running.

I have left the nugets as per sample:
`<package id="dbup" version="3.3.5" targetFramework="net452" />
  <package id="dbup-firebird" version="3.3.5" targetFramework="net452" />
  <package id="FirebirdSql.Data.FirebirdClient" version="4.5.2.0" targetFramework="net452" />`

I'm running Firebird 3, 64bit, 3.0.0.32483.

I have a populated db and I'm trying to run 1 script:
`CREATE TABLE Testing1 (ID int NOT NULL primary key);`

The schema table is being created and the Testing1 table is being created, then it fails on:
> at DbUp.Engine.UpgradeEngine.PerformUpgrade() in C:\projects\dbup\src\DbUp\Engine\UpgradeEngine.cs:line 73

> No message for error code 336397285 found.
>No message for error code 336068862 found. ---> unsuccessful metadata update
>No message for error code 336397285 found.
>No message for error code 336068862 found.

I've tried commenting out some settings, I always get the Testing1 table created, with this line in place:
`.WithTransactionPerScript()`
I don't get the schema table created, if I comment out this line, I do...
including `.WithoutTransaction()` has no effect.

Any ideas before I go hunting into the DBup code? What version of Firebird was this Sample working for?

**Comment by Timothyoverton** [comment]
Anyone?

**Comment by Timothyoverton** [comment]
This issue can also be resolved with my fix here: https://github.com/DbUp/DbUp/pull/250

**Comment by hhindriks** [comment]
In intend to fix any issues with the FirebirdSampleApplication and submit a PR for it soon. Currently I need to comment out both 
` 
.WithTransactionPerScript() and 
.WithTransaction()
`
for the application to run through.

**Comment by Jeern** [comment]
@mjauernig  and @sungam3r  This was fixed by my PR https://github.com/DbUp/DbUp/pull/692

I have tested that the SampleApplication works. However

1. WithTransaction is no longer allowed (AllowedTransactionMode)
2. WithTransactionPerScript did not work before my fix, but works now.
3. I need to upgrade the FirebirdSql.Data.FirebirdClient to the latest to make it work for FIrebird versions newer than 2.5 (because the old Nuget package does not support Wire encryption)

I will provide af PR for nr. 3 - after that this Issue can be closed
